lib/blocktree/blocktree.go

```diff
@@ -87,17 +76,16 @@ func (bt *BlockTree) AddBlock(header *types.Header, arrivalTime time.Time) error
 		return errUnexpectedNumber
 	}
 
-	n = &node{
+	n := &node{
 		hash:        header.Hash(),
 		parent:      parent,
 		children:    []*node{},
 		number:      number,
 		arrivalTime: arrivalTime,
 	}
+
 	parent.addChild(n)
```

Comment:
Want to make sure I understand here, so we are okay with concurrent calls to addChild here because whichever acquires the lock will perform the replace, then that state change to the leafMap would invalidate the other replace requests if its the same parent. Is this right? 

Sorry if that was confusing, I'm trying to reason about moving the lock from add into replace

Reply:
sorry the lock was not supposed to be removed from AddBlock lol, but in general the functions that are exported need to lock and the internal ones don't as they should only be called by functions that lock

Reply:
Ahhh okay that answers the question then lol, thanks!